staging/src/k8s.io/api/bulk/v1alpha1/types.go

```diff
@@ -0,0 +1,216 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkRequest contains a single request to bulk api.
+type BulkRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Opaque identifier used to match requests and responses.
+	// +optional
+	RequestID string `json:"requestId,omitempty" protobuf:"bytes,1,opt,name=requestId"`
+
+	// Starts new watch.
+	// +optional
+	Watch *WatchOperation `json:"watch,omitempty" protobuf:"bytes,2,opt,name=watch"`
+
+	// Starts new watch.
+	// +optional
+	WatchList *WatchListOperation `json:"watchList,omitempty" protobuf:"bytes,3,opt,name=watchList"`
+
+	// Stops existing watch.
+	// +optional
+	StopWatch *StopWatchOperation `json:"stopWatch,omitempty" protobuf:"bytes,4,opt,name=stopWatch"`
+
+	// Gets list of items.
+	// +optional
+	List *ListOperation `json:"list,omitempty" protobuf:"bytes,5,opt,name=list"`
+
+	// Gets single item.
+	// +optional
+	Get *GetOperation `json:"get,omitempty" protobuf:"bytes,6,opt,name=get"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient:nonNamespaced
+
+// BulkResponse contains a single response from bulk api.
+type BulkResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Opaque identifier propagated from the request.
+	// Null for messages issued by server (WatchEvent, WatchStopped triggered by server)
+	// +optional
+	RequestID *string `json:"requestId,omitempty" protobuf:"bytes,1,opt,name=requestId"`
+
+	// Contains error details when request was failed.
+	// +optional
+	Failure *metav1.Status `json:"failure,omitempty" protobuf:"bytes,2,opt,name=failure"`
+
+	// Sent when watch was manually stopped.
+	// +optional
+	WatchStopped *WatchStopped `json:"watchStopped,omitempty" protobuf:"bytes,3,opt,name=watchStopped"`
+
+	// Sent when new watch was started.
+	// +optional
+	WatchStarted *WatchStarted `json:"watchStarted,omitempty" protobuf:"bytes,4,opt,name=watchStarted"`
+
+	// Contains single watch event.
+	// +optional
+	WatchEvent *BulkWatchEvent `json:"watchEvent,omitempty" protobuf:"bytes,5,opt,name=watchEvent"`
+
+	// Contains single object.
+	// +optional
+	GetResult *GetResult `json:"getResult,omitempty" protobuf:"bytes,6,opt,name=getResult"`
+
+	// Contains single object.
+	// +optional
+	ListResult *ListResult `json:"listResult,omitempty" protobuf:"bytes,7,opt,name=listResult"`
+}
+
+// GroupVersionResource identifies resource.
+type GroupVersionResource struct {
+
```

Comment:
For consistency with other "types.go" files in different places, can you please remove empty lines right after type definitions?